lib/commons/text/unicode.js

```diff
@@ -91,6 +91,7 @@ function getUnicodeNonBmpRegExp() {
 		'\u25A0-\u25FF' + // Geometric Shapes
 		'\u2600-\u26FF' + // Misc Symbols
 		'\u2700-\u27BF' + // Dingbats
+		'\uE000-\uF8FF' + // Private Use
```

Comment:
Going with definition from here:

>  Does Unicode have private-use characters?

> A: Yes. There are three ranges of private-use characters in the standard. The main range in the BMP is U+E000..U+F8FF, containing 6,400 private-use characters. That range is often referred to as the Private Use Area (PUA). But there are also two large ranges of supplementary private-use characters, consisting of most of the code points on Planes 15 and 16: U+F0000..U+FFFFD and U+100000..U+10FFFD. Together those ranges allocate another 131,068 private-use characters. Altogether, then, there are 137,468 private-use characters in Unicode.

It looks like we are only ignoring ones in BMP and not the supplementary private-use characters. Shouldn't we consider the supplementary ones too?

Reply:
I thought about it, but I haven't been able to find a 100% reliable way to test them. I wasn't able to find anything directly on the internet, but was able to piece together how we might be able to using surrogate pairs of the [high](https://www.unicode.org/charts/PDF/UD800.pdf) and [low](https://www.unicode.org/charts/PDF/UDC00.pdf) surrogates together. Theoretically this regex should catch them, but I can't be sure.

```
/[\uDB80-\uDBBF][\uDC00-\uDFFD]/
```

Reply:
Given that nothing in that range is likely to be something Dragon knows how to handle, I think it's a pretty safe bet to just exclude the entire range.